For #25025 - Integrate Pocket sponsored stories

Mugurell opened the pull request:
~Needs https://github.com/mozilla-mobile/android-components/pull/12070~
~Needs https://github.com/mozilla-mobile/android-components/pull/12183~

~Based on https://github.com/mozilla-mobile/fenix/pull/24892.~

### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Mugurell commented:
Rebased to fix conflicts.

Mugurell commented:
In the meantime the design changed a little. Figma link available on Jira.
Updated the code now to comply.

Mugurell commented on app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt:
```diff
@@ -114,21 +118,78 @@ fun PocketStory(
+            Icon(
+                painter = painterResource(id = R.drawable.pocket_star_stroke),
+                contentDescription = null,
+                tint = FirefoxTheme.colors.iconSecondary,
```
This and the text color below are slightly different than on Figma to not have to add a custom color.
Asked Sam from UX for validating any one approach.

Mugurell replied:
Apparently the slightly different color was just a design quirk.
Okayed to use the same `textSecondary` / `iconSecondary` colors.

mcarare commented on app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt:
```diff
@@ -54,10 +55,26 @@ sealed class AppAction : Action {
+     */
+    object PocketStoriesClean : AppAction()
+    /**
+     * Replace the current list of Pocket sponsored stories.
```
nit: `Replaces` to follow the pattern in other actions doc.

mcarare commented on app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt:
```diff
@@ -54,10 +55,26 @@ sealed class AppAction : Action {
+     */
+    data class PocketStoriesShown(val storiesShown: List<PocketStory>) : AppAction()
+    /**
+     * Clean all in-memory data about Pocket stories and categories.
```
nit: `Cleans` to follow the pattern in other actions doc.

mcarare commented on app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesViewHolder.kt:
```diff
@@ -23,7 +23,7 @@ import androidx.compose.ui.unit.dp
-import mozilla.components.service.pocket.PocketRecommendedStory
+import mozilla.components.service.pocket.PocketStory.PocketRecommendedStory
 import org.mozilla.fenix.R
 import org.mozilla.fenix.R.string
```
Not from this change, but we probably do not want this import.

Mugurell commented:
Thank you Mihai!
Addressed now your comments and rebased. Should be good to launch tomorrow along with https://github.com/mozilla-mobile/android-components/pull/12183.

Mugurell commented:
Restarting CI.

Mugurell commented:
`toolchain-linux64-android-gradle-dependencies ` failed again because of:
```
Execution failed for task ':app:packageDebug'.
> Multiple task action failures occurred:
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Self-suppression not permitted
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Java heap space
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Self-suppression not permitted
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Java heap space
```

Restarting CI.
